### PR TITLE
Implementation for Admin / Super Admin Roles

### LINKED ISSUE #1185: Bug Report - Correctly Implement Admin / Super Admin Roles
**Describe the bug**

This Admin/Super Admin functionality is not currently implemented correctly. This is how it should work:

**Background**

Say for example that a chain of non-profit community centers with branches in multiple neighborhoods decides to use Talawa. They have a large membership across the region but each member tends to use only their local community center. From time to time members move into new neighborhoods where they decide to switch their membership to their new local branch.

The VP of community relations would want to be the Super Admin as they would be able to add new branches as they were established. The VP would then delegate the management of community relations to each branch manager who would then become an Admin for their local membership.

The VP doesn’t want the branch managers to manage the members who have joined other branches, just the branch they are running.

It is for this reason that we have two types of administrators. Admins and Super Admins.

1. **Admin**: These are members who use use Talawa Admin to manage the people in an organization. This would also include calendars and the organization's news feed.
    1. Admins can only manage a single organization and cannot access any data from other organizations. Expanding this capability is a long term goal.
    1. Admins can promote Users and Members to Admin status for their organization only
1. **Super Admin:** These are members who use use Talawa Admin to manage all organizations in a community.
    1. Super Admins can promote Users and Members to Admin or Super Admin status.
    1. Super Admins have access to all the privileges that an Admin may have over the Admin's organization.

Like other people, an administrator may or may not have joined an organization in the mobile app.

**Expected behavior**

The API needs to enforce these rules.

**Actual behavior**

Depending on the PR.

1. Admins can manage multiple organizations
2. Admins can freely upgrade and downgrade themselves to SuperAdmin status

**Screenshots**
N/A

**Additional details**
Read the core concepts document for more details
- https://docs.talawa.io/docs/applications/overview
- https://github.com/PalisadoesFoundation/talawa-admin/issues/727

### PR DESCRIPTION
**Pull request by KrutikaBhatt**
**What kind of change does this PR introduce?**

This PR provides solution for developing functionalities for Admin/Super Admin Roles. The following changes are made : 
1. `createAdmin` - In place of superAdmin check, I have added a check for admin or super Admin
2. `updateOrgUserType` - There is already a  mutation that allows a Super Admin to promote any user to admin and change its roles. With this mutation, In the admin side, the admin can also do the same but with only the members of its own organization.

Please Note : _**The PR is still in progress**_

**Issue Number:#1185**

Fixes #1185

**Did you add tests for your changes?** Not yet, Once the changes got approved by the talawa-admin side, I will add the tests for the same

**If relevant, did you update the documentation?**

**Does this PR introduce a breaking change?** No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)? Yes**

### CODE REVIEW

**Comment by KrutikaBhatt** [comment]
@prathamesh-mutkure, please check the new additions in the `talawa-api`. This should help to create the changes in the `talawa-admin` side also. Incase of any updates or modifications, do let me know

**Comment by palisadoes** [comment]
Sorry for the delay in assigning reviewers. @xoldyckk @Kevoniat Please review.

File: src/typeDefs/inputs.ts [hunk 1]
```diff
@@ -285,6 +285,12 @@ export const inputs = gql`
     id: ID
   }
 
+  input UpdateOrgUserTypeInput {
+    userType: String
+    id: ID
+    organizationId: ID
```
Comment (xoldyckk) [inline src/typeDefs/inputs.ts:291]: Make all these arguments required, else it doesn't make to trigger the mutation on client side

**Comment by palisadoes** [comment]
@KrutikaBhatt Please make the suggested changes so that we can close this PR

**Comment by palisadoes** [comment]
Closing due to inactivity

**Comment by kirtanchandak** [comment]
can I take this issue?

**Pull request closed without merging**